app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt

```diff
@@ -26,19 +67,140 @@ class SplashActivityTest {
     SplashActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
   )
 
-  @Before
-  fun setUp() {
-    Intents.init()
+  @Test
+  fun testSplashActivity_initialOpen_routesToOnboardingActivity() {
+    activityTestRule.launchActivity(null)
+    intended(hasComponent(OnboardingActivity::class.java.name))
   }
 
   @Test
-  fun testSplashActivity_initialOpen_routesToHomeActivity() {
-    activityTestRule.launchActivity(null)
-    intended(hasComponent(ProfileActivity::class.java.name))
+  fun testSplashActivity_secondOpen_routesToChooseProfileActivity() {
+    simulateAppAlreadyOnboarded()
+    launch(SplashActivity::class.java).use {
+      intended(hasComponent(ProfileActivity::class.java.name))
+    }
   }
 
-  @After
-  fun tearDown() {
-    Intents.release()
+  private fun simulateNewAppInstance() {
+    // Simulate a fresh app install by clearing any potential on-disk caches using an isolated onboarding flow controller.
+    createTestRootComponent().getOnboardingFlowController().clearOnboardingFlow()
```

Comment:
@BenHenning In this `clearOnboardingFlow` cannot be used now as it was removed from @nikitamarysolomanpvt 's PR as per your comments. Now, the splash screen works in following ways:
1. if the user has already onboarded -> open `ProfileActivity` else open `OnboardingFlowActivity`.

Now to test this correctly, we need to clear the cache of onboarding flow, but that function has been removed from the OnboardingFlowController file. So how can I still do that cache-clear thing in app module test cases.

Reply:
Can we write something similar to ProfileTestHelper in which there can be a function which helps in clearing this cache?

Reply:
Yes, I think a test helper is a good way to solve this problem rather than exposing a test-only API in the domain controller. In such a helper, you could either:

1) Delete the actual file from the filesystem

2) Create a new persistent cache store object and use its clear method

In either case, you should move the name of the cache to a dagger constant so that you can share & inject it both in the domain controller and test helper. I suggest doing this as a dedicated PR to keep things clean.

Reply:
Created PR #670 for this.

Reply:
Done. via just recreating the root component in test file